app/tests/util/webhooks.py

```diff
@@ -20,42 +26,75 @@ def get_wallet_id_from_async_client(client: RichAsyncClient) -> str:
 async def check_webhook_state(
     client: RichAsyncClient,
     topic: CloudApiTopics,
-    filter_map: Dict[str, Optional[str]],
+    state: str,
+    filter_map: Optional[Dict[str, str]] = None,
     max_duration: int = 60,
     lookback_time: int = 1,
-) -> bool:
+) -> Dict[str, Any]:
     assert max_duration >= 0, "Poll duration cannot be negative"
 
     wallet_id = get_wallet_id_from_async_client(client)
 
     listener = SseListener(wallet_id, topic)
+    bound_logger = logger.bind(body={"wallet_id": wallet_id, "topic": topic})
 
-    desired_state = filter_map["state"]
+    # Retry logic in case of disconnect errors (don't retry on timeout errors)
+    event = None
+    attempt = 0
+    max_tries = 2
+    delay = 0.5
 
-    other_keys = dict(filter_map)
-    other_keys.pop("state", None)
+    while not event and attempt < max_tries:
```

Comment:
@cl0ete @wdbasson Here is the retry logic implementation.

Currently tries max 2 times. i.e. just one retry, and only if there's an HTTP Error. Timeouts won't retry.